src/pages/puzzles/list/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { GetServerSideProps } from 'next';
+import { getAllPuzzleInstances } from '../../../services/puzzleInstance';
+import { Header, SearchAndFilter } from '../../../components/Global';
+import { CardGrid } from '../../../components/App';
+import styles from '../../../styles/pages/PuzzleList.module.scss';
+import { PuzzleInstance } from '@prisma/client';
+
+type PuzzleListTypes = {
+  puzzleInstances: PuzzleInstance[];
+};
+
+const PuzzleList = ({ puzzleInstances }: PuzzleListTypes) => {
+  const [searchNFilter, setSearchNFilter] = useState<{
+    searchText: string;
+    filterFields: { EASY: boolean; MEDIUM: boolean; HARD: boolean };
+  }>({
+    searchText: '',
+    filterFields: { EASY: true, MEDIUM: true, HARD: true }
+  });
+  return (
+    <>
+      <Header />
+      <div className={styles.searchAndFilter}>
+        <SearchAndFilter setSearchNFilterVal={setSearchNFilter} />
+      </div>
+
+      <div className={styles.cardGrid}>
+        <CardGrid
+          cardList={puzzleInstances
+            .filter(instance => {
+              if (searchNFilter.searchText) {
+                return (
+                  searchNFilter.filterFields[instance.puzzle.difficulty] ===
+                    true &&
+                  instance.puzzle.name.includes(searchNFilter.searchText)
+                );
+              } else {
+                return (
+                  searchNFilter.filterFields[instance.puzzle.difficulty] ===
+                  true
+                );
+              }
+            })
+            .map(instance => {
+              //return a card prop object for each instance
+              return {
+                name: instance.puzzle.name,
+                content: instance.puzzle.content,
+                difficulty: instance.puzzle.difficulty,
+                buttonActions: [
+                  {
+                    text: 'View On Map',
+                    style: 'secondary',
+                    // action?: () => any, TODO: pass the instanceId to the map page, so user don't to filter again
+                    link: '/puzzles/map'
+                  },
+                  {
+                    text: 'Solve Online',
+                    style: 'primary',
+                    link: `/puzzles/[${instance.id}]`
```

Comment:
I don't think we're supposed to put square brackets around the id here - does this work?

Reply:
I can't test it yet. There's a function in getServerSideProps that depends on #85.
But, yah, the square brackets doesn't make sense. I removed it